examples/counter-contract/src/lib.rs

```diff
@@ -1,36 +1,117 @@
 // Do not link against libstd (i.e. anything defined in `std::`)
-#![no_std]
-#![feature(alloc_error_handler)]
+#![cfg_attr(not(test), no_std)]
```

Comment:
We're not actually dependent on `std` for the code we emit via `#[miden_test]`/`#[miden_test_suite]` AFAICT, so do we actually need to make this conditionally compiled?